### PR TITLE
[Notion] Change GC complexity

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

Change the big O complexity of the GC notion_pages fetching algorithm from O(N^2) to O(N).

### CODE REVIEW

File: connectors/src/connectors/notion/temporal/activities.ts [hunk 1]
```diff
@@ -972,44 +972,34 @@ async function findResourcesNotSeenInGarbageCollectionRun(
...
+      skipReason: p.skipReason || null,
+    }));
 
-    pagesNotSeenInGarbageCollectionRun.push(...pages);
-    offset += pageSize;
-  }
+  pagesNotSeenInGarbageCollectionRun.push(...pages);
 
-  offset = 0;
+  let offset = 0;
```
Comment (fontanierh) [inline connectors/src/connectors/notion/temporal/activities.ts:1002]: I think we want the exact same for databases below no ?

**Review by fontanierh — APPROVED** [review]
Looks great but I think we want to do the same below where we have the exact same patter

**Comment by lasryaric** [comment]
We could but they are not problematic for now.

**Comment by lasryaric** [comment]
And I would like to see how this behaves before moving forward. WDYT?

**Comment by fontanierh** [comment]
Good for me @lasryaric

**Pull request merged**